### PR TITLE
Only configure HockeyApp when the app is not prelaunched

### PR DESCRIPTION
**Pull request by achocron**

### CODE REVIEW

File: UWP/HockeyAppDemoUwp/cs/App.xaml.cs [hunk 1]
```diff
@@ -31,20 +31,33 @@ sealed partial class App : Application
...
+            this.Resuming += App_Resuming;
             this.InitializeComponent();
             this.Construct();
         }
 
+        private void App_Resuming(object sender, object e)
+        {
+            // If an app is opened after being prelaunched, OnLaunched is not called and
+            // the Resuming event gets fired.
+            ConfigureHockeyApp();
```
Comment (izikl) [inline UWP/HockeyAppDemoUwp/cs/App.xaml.cs:51]: Is there a scenario where App_Resuming is called and OnLaunched or OnActivated are not?

File: UWP/HockeyAppDemoUwp/cs/project.lock.json [hunk 2]
```diff
@@ -11695,7 +11695,7 @@
       ]
     },
     "Microsoft.CSharp/4.0.0": {
-      "sha512": "/q7Wo1eEf1Nv2cuOO5h1F/MzEM0xn0MoV2kb1YpgDZFYZ/y5gkPYUA0Ws6mVADJKiMWnl2/RXb87roVhUy0QXA==",
```
Comment (izikl) [inline UWP/HockeyAppDemoUwp/cs/project.lock.json:11698]: I don't think this file should be checked in. 
http://stackoverflow.com/questions/38065611/what-is-project-lock-json

Can you delete it, and filter it out in .gitignore?

File: UWP/HockeyAppDemoUwp/cs/App.xaml.cs [hunk 3]
```diff
@@ -128,5 +146,23 @@ void OnNavigationFailed(object sender, NavigationFailedEventArgs e)
 
         // Add any application contructor code in here.
         partial void Construct();
+
+        private bool HockeyAppConfigured = false;
+        private void ConfigureHockeyApp()
+        {
+            if (HockeyAppConfigured)
```
Comment (izikl) [inline UWP/HockeyAppDemoUwp/cs/App.xaml.cs]: Can we add this check in the SDK? After all we always expect configure to happen once so why not enforcing it?

**Review by izikl — CHANGES_REQUESTED** [review]
Overall looks good. Some ideas inside.

File: UWP/HockeyAppDemoUwp/cs/App.xaml.cs (hunk 3 above)
Comment (achocron) [reply UWP/HockeyAppDemoUwp/cs/App.xaml.cs]: There is the method CheckForInitialization(), but it throws an exception if it hasn't been configured, and I didn't want to throw an exception every time the app is resumed

File: UWP/HockeyAppDemoUwp/cs/App.xaml.cs (hunk 1 above)
Comment (achocron) [reply UWP/HockeyAppDemoUwp/cs/App.xaml.cs:51]: If App_Resuming is called then I believe either OnLaunched or OnActivated has been called beforehand (though not necessarily immediately beforehand)

**Pull request closed without merging**